test/Oscoin/Test/P2P/Handshake.hs

```diff
@@ -125,6 +147,26 @@ prop_secureHandshake Dict = withTests 1 . property $ do
             Left Noise.DecryptionError -> success
             _                          -> failure
 
+prop_secureRejectsSelf :: forall c. Dict (IsCrypto c) -> Property
+prop_secureRejectsSelf Dict = withTests 1 . property $ do
+    (pk, sk) <- evalIO (Crypto.generateKeyPair @c)
```

Comment:
Tip: if you want a predictable key-pair, you can use the `FastRandomBytes` abstraction from `Oscoin.Test.Crypto.PubKey.Arbitrary` to write something like:

```
seed <- forAll (Gen.word64 Range.linearBounded)
(pk, sk) <- forAll (withFastRandomBytes (SplitMix.mkSMGen seed) generateKeyPair)
```